src/mir/type-of.rkt

```diff
@@ -157,13 +158,13 @@
   [; ref
    (type-of/Place Γ Place Ty)
    ------------------------------------------
-   (type-of/Rvalue Γ (ref Lt () Place) (& Lt Ty))
+   (type-of/Rvalue Γ (ref Lt () Place) (rigid-ty (ref ()) (Lt Ty)))
```

Comment:
`   (type-of/Rvalue Γ (ref Lt () Place) (user-ty (& Lt Ty)))`, maybe?

Reply:
This won't work since we currently can't turn a `Ty` into a `UserTy`. Originally, there existed a `&` metafunction that created reference types.

Reply:
Oh, I see, yes. Hmm =) Maybe we want a `ref-ty` helper or something, but this is fine.